aws/data_source_aws_ami_test.go

```diff
@@ -38,7 +38,6 @@ func TestAccAWSAmiDataSource_natInstance(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "image_type", "machine"),
 					resource.TestCheckResourceAttr(resourceName, "most_recent", "true"),
 					resource.TestMatchResourceAttr(resourceName, "name", regexp.MustCompile("^amzn-ami-vpc-nat")),
-					resource.TestCheckResourceAttr(resourceName, "owner_id", "137112412989"),
```

Comment:
Unnecessary since `image_owner_alias` is checked to be `amazon`. Also, not consistent across partitions and potentially over time.

Reply:
Rather than removing these, we should verify that something was set for the attribute to prevent future regressions should a `d.Set()` be errantly removed, e.g.

```
// preferably
// maybe could be its own "testAccTestMatchResourceAttrAccountID(resourceName, attribute) helper
// and lintable!
resource.TestMatchResourceAttr(resourceName, "owner_id", regexp.MustCompile(`\d{12}`)),

// otherwise
resource.TestCheckResourceAttrSet(resourceName, "owner_id")
```